pgmpy/tests/test_models/test_BayesianNetwork.py

```diff
@@ -1807,23 +1805,92 @@ def test_missing_scheme_sample(self):
         samples = self.con_model.simulate(n_samples=1000)
         self.assertFalse(samples.isnull().values.any())
 
-        samples = self.con_model.simulate(
-            n_samples=1000, include_missing=True, missing_prob=0.9
+        cpd = TabularCPD(
+            "Z*",
+            2,
+            [[0.5], [0.5]],
         )
-        self.assertTrue(samples.isnull().values.any())
+        samples = self.con_model.simulate(n_samples=1000, missing_prob=cpd)
+        missing_fraction = samples["Z"].isnull().mean()
+        self.assertGreaterEqual(missing_fraction, 0.45)
+        self.assertLessEqual(missing_fraction, 0.55)
+        self.assertFalse(samples.drop(columns=["Z"]).isnull().values.any())
 
-        miss_columns = ["X", "U"]
-        samples = self.con_model.simulate(
-            n_samples=1000,
-            seed=42,
-            include_missing=True,
-            missing_prob=0.9,
-            missing_columns=miss_columns,
+        cpd = TabularCPD(
+            "Z*", 2, [[0.5, 0.5, 0.5, 0.5], [0.5, 0.5, 0.5, 0.5]], ["X", "Y"], [2, 2]
+        )
+        samples = self.con_model.simulate(n_samples=1000, missing_prob=cpd)
+        grouped = samples.groupby(["X", "Y"], observed=False)["Z"]
+        for (x, y), group in grouped:
+            missing_fraction = group.isnull().mean()
+            expected_missing_fraction = cpd.values[0][x, y]
+            self.assertAlmostEqual(
+                missing_fraction, expected_missing_fraction, delta=0.15
```

Comment:
Would suggest increasing the sample size to reduce the delta required here. Would ideally want to keep the delta under 0.1